firewood/src/manager.rs

```diff
@@ -260,30 +291,50 @@ impl RevisionManager {
     }
 }
 
-impl RevisionManager {
+impl<T: RootStore> RevisionManager<T> {
     pub fn add_proposal(&self, proposal: ProposedRevision) {
         self.proposals.lock().expect("poisoned lock").push(proposal);
     }
 
+    /// View the database at a specific time.
+    /// To view the database at a specific time involves a few steps:
+    /// 1. Try to find it in committed revisions.
+    /// 2. Try to find it in proposals.
+    /// 3. Try to find it in `RootStore`.
     pub fn view(&self, root_hash: HashKey) -> Result<ArcDynDbView, RevisionManagerError> {
-        // First try to find it in committed revisions
+        // 1. Try to find it in committed revisions.
         if let Ok(committed) = self.revision(root_hash.clone()) {
             return Ok(committed);
         }
 
-        // If not found in committed revisions, try proposals
+        // 2. Try to find it in proposals.
         let proposal = self
             .proposals
             .lock()
             .expect("poisoned lock")
             .iter()
             .find(|p| p.root_hash().as_ref() == Some(&root_hash))
-            .cloned()
-            .ok_or(RevisionManagerError::RevisionNotFound {
-                provided: root_hash,
-            })?;
+            .cloned();
 
-        Ok(proposal)
+        if let Some(v) = proposal {
+            return Ok(v);
+        }
```

Comment:
I would have combined this with the previous block, something like:
```
        if let Some(proposal) = self.proposals.lock()...cloned() {
            return Ok(proposal);
        }
```

Avoids `v`. If you want to keep your old code, at rename `v` and shadow as `proposal`:
```suggestion
        if let Some(proposal) = proposal {
            return Ok(proposal);
        }
```